text/0000-rust-lang-crates.md

```diff
@@ -0,0 +1,231 @@
+ownership can be transferred externally.
+
+## Advertising
+
+Part of the reason for having rust-lang crates is to have a clear, short list of
+libraries that are broadly useful, vetted and maintained. But where should this
+list appear?
+
+This RFC doesn't specify the complete details, but proposes a basic direction:
+
+- The crates in rust-lang should appear in the sidebar in the core rustdocs
+  distributed with Rust, along side the standard library. (For nightly releases,
+  we should include the nursery crates as well.)
+
+- The crates should also be published on crates.io, and should somehow be
+*badged*. But the design of a badging/curation system for crates.io is out of
+scope for this RFC.
+
+## Plan for existing crates
+
+There are already a number of non-`std` crates in rust-lang. Below, we give the
+full list along with recommended actions:
+
+### Transfer ownership
+
+Please volunteer if you're interested in taking one of these on!
+
+- rlibc
+- semver
+- threadpool
+
+### Move to rust-lang-nursery
+
+- bitflags
+- getopts
+- glob
+- libc
+- log
+- rand (note, @huonw has a major revamp in the works)
+- regex
+- rustc-serialize (but will likely be replaced by serde or other approach eventually)
+- tempdir (destined for `std` after reworking)
+- uuid
+
+### Move to rust-lang-deprecated
+
+- fourcc: highly niche
+- hexfloat: niche
+- num: this is essentially a dumping ground from 1.0 stabilization; needs a complete re-think.
+- term: API needs total overhaul
+- time: needs total overhaul destined for std
+- url: replaced by https://github.com/servo/rust-url
+
+# Drawbacks
+
+The drawbacks of this RFC are largely social:
+
+* Emphasizing rust-lang crates may alienate some in the Rust community, since it
+  means that certain libraries obtain a special "blessing". This is mitigated by
+  the fact that these libraries also become owned by the community at large.
+
+* On the other hand, requiring that ownership/governance be transferred to the
+  library subteam may be a disincentive for library authors, since they lose
+  unilateral control of their libraries. But this is an inherent aspect of the
+  policy design, and the vastly increased visibility of libraries is likely a
+  strong enough incentive to overcome this downside.
+
+# Alternatives
+
+The main alternative would be to not maintain other crates under the rust-lang
+umbrella, and to offer some other means of curation (the latter of which is
+needed in any case).
+
+That would be a missed opportunity, however; Rust's governance and maintenance
+model has been very successful so far, and given our minimalistic plans for the
+standard library, it is very appealing to have *some* other way to apply the
+full Rust community in taking care of additional crates.
+
+# Unresolved questions
+
+Part of the maintenance standard for Rust is the CI infrastructure, including
+bors/homu. What level of CI should we provide for these crates, and how do we do it?
```

Comment:
I believe this concern is not so much motivated by automation to merge crates but other aspects such as:

* Running all these builds every night
* Platform coverage -- are they all guaranteed to work on 1st tier platforms? Second tier? Do we have automation to prove it?
* Do we gate releases on these crates working? How is that all set up?

The PR volume on external crates is generally low enough that homu isn't necessary most of the time (unlike for the rust-lang/rust repo)